libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceUtil.kt

```diff
@@ -37,31 +39,35 @@ object MapboxDistanceUtil {
         context: Context,
         locale: Locale
     ): FormattedDistanceData {
-        return when (distanceInMeters) {
-            !in 0.0..Double.MAX_VALUE -> {
-                formatDistanceAndSuffixForSmallUnit(
-                    0.0,
-                    roundingIncrement,
-                    unitType,
-                    context,
-                    locale
-                )
-            }
-            in 0.0..smallDistanceUpperThresholdInMeters -> {
-                formatDistanceAndSuffixForSmallUnit(
-                    distanceInMeters,
-                    roundingIncrement,
-                    unitType,
-                    context,
-                    locale
-                )
-            }
-            in smallDistanceUpperThresholdInMeters..mediumDistanceUpperThresholdInMeters -> {
-                formatDistanceAndSuffixForLargeUnit(distanceInMeters, 1, unitType, context, locale)
-            }
-            else -> {
-                formatDistanceAndSuffixForLargeUnit(distanceInMeters, 0, unitType, context, locale)
-            }
+        return when (getFormattingRange(distanceInMeters, unitType)) {
+            FormattingRange.INVALID -> formatDistanceAndSuffixForSmallUnit(
+                0.0,
+                roundingIncrement,
+                unitType,
+                context,
+                locale
+            )
+            FormattingRange.SMALL -> formatDistanceAndSuffixForSmallUnit(
+                distanceInMeters,
+                roundingIncrement,
+                unitType,
+                context,
+                locale
+            )
+            FormattingRange.MEDIUM -> formatDistanceAndSuffixForLargeUnit(
+                distanceInMeters,
+                1,
```

Comment:
out of scope: do you know why does `roundingIncrement` is ignoring for Medium and Large values?

Reply:
I can only guess. I think it only makes sense for small units since their values are large. For example, we don't care if it's 451 or 453 meters, while 11 vs 12 km is quite a difference. If the user passes `roundingIncrement = 5` meaning that they want to display 451 meter as 450 meters, it probably doesn't mean that they also want to display 5 km instead of 4 km. That's just my guess but I'd say the docs are not clear enough (we should specify that it's only valid for small distances).

Reply:
yeah, that makes sense. The API seems a bit confusing:  it provides the opportunity to set this value, but later one is ignored. 😞